File: common/src/main/java/org/bread_experts_group/breadlib/test/server/TasksServerTest.java
package org.bread_experts_group.breadlib.test.server;

import org.bread_experts_group.breadlib.task.TaskManager;
import org.bread_experts_group.breadlib.task.network.NetworkTask;
import org.bread_experts_group.breadlib.test.ServerboundPacketTest;

public class TasksServerTest {
	public static void networkTest() {
		TaskManager.newTask(NetworkTask.class, task -> task.addServerbound(
				ServerboundPacketTest.class,
				ServerboundPacketTest.TYPE,
				ServerboundPacketTest.STREAM_CODEC,
				ServerboundPacketTest::handleServerbound
		));
	}
}

File: common/src/main/kotlin/org/bread_experts_group/breadlib/test/server/TasksServerTest.kt
package org.bread_experts_group.breadlib.test.server

import org.bread_experts_group.breadlib.task.TaskManager
import org.bread_experts_group.breadlib.task.network.NetworkTask
import org.bread_experts_group.breadlib.test.ServerboundPacketTest

object TasksServerTest {
	fun networkTest() {
		TaskManager.newTask { task: NetworkTask ->
			task.addServerbound(
				ServerboundPacketTest::class.java,
				ServerboundPacketTest.TYPE,
				ServerboundPacketTest.STREAM_CODEC,
				ServerboundPacketTest::handleServerbound
			)
		}
	}
}
